content/blog/2024/2024-02-13-apache-http-client-classic-apis.md

```diff
@@ -0,0 +1,776 @@
+      String responseBody = httpClient.execute(httpHost, httpGetRequest, handler);
+
+      return responseBody;
+    } catch (Exception e) {
+      throw new RequestProcessingException("Failed to get paginated users.", e);
+    }
+  }
+}  
+```
+
+The code defines `getPaginatedUsers()` method to retrieve a list of users from an external API, specified by the request parameters map.  The `requestParameters` are mapped into a list of `NameValuePairs`. Then we create `HttpGet` instance, representing the GET request and call HttpClient's `execute()` method. The response body returned by the server is stored in the responseBody variable.
+
+Here is a test case to verify this functionality:
+
+```java
+public class UserSimpleHttpRequestHelperTests extends BaseClassicExampleTests {
+
+  private UserSimpleHttpRequestHelper userHttpRequestHelper =
+      new UserSimpleHttpRequestHelper();
+
+  /** Execute get paginated request. */
+  @Test
+  void executeGetPaginatedRequest() {
+    try {
+      // prepare
+      Map<String, String> params = Map.of("page", "1");
+
+      // execute
+      String responseBody = userHttpRequestHelper.getAllUsers(params);
+
+      // verify
+      assertThat(responseBody).isNotEmpty();
+    } catch (Exception e) {
+      Assertions.fail("Failed to execute HTTP request.", e);
+    }
+  }
+}
+```
+In test method `executeGetPaginatedRequest()`, we populate the request parameter (page=1) and execute an HTTP GET request to retrieve the first page of paginated users and verify the response.
+
+### Executing HTTP GET Request to Get Specific Record
+
+Let's now see how to execute HTTP GET request to get a specific user record using a response handler:
+
+```java
+public class UserSimpleHttpRequestHelper extends BaseHttpRequestHelper {
+ 
+ /** Gets user for given user id. */
+ public String getUser(long userId) throws RequestProcessingException {
+    
+    try (CloseableHttpClient httpClient = HttpClients.createDefault()) {
+      // Create request
+      HttpHost httpHost = HttpHost.create("https://reqres.in");
+      HttpGet httpGetRequest 
+        = new HttpGet(new URIBuilder("/api/users/" + userId).build());
+      
+      // Create a response handler
+      BasicHttpClientResponseHandler handler = new BasicHttpClientResponseHandler();
+      String responseBody = httpClient.execute(httpHost, httpGetRequest, handler);
+
+      return responseBody;
+    } catch (Exception e) {
+      throw new RequestProcessingException(
+          MessageFormat.format("Failed to get user for ID: {0}", userId), e);
+    }
+  }
+}
+
+```
+
+In this example, the `getUser()` method retrieves a user by its id. As we have learned in `getAllUsers()` code example, in this case also, we create a `HttpGet` request object, a `HttpHost` object  and response handler. Then we call `execute()` method on the client. Then we obtain the response in string form.
+
+Now Let's see how to call this functionality.
```

Comment:
Overal, we repeat "Let’s see how to do it:" and "Now Let’s see how to call this functionality." quite a lot. Could this be phrased differently? For example by leaving it out?

Reply:
Great find! 
I have tried my best to avoid repetition.